src/v/security/audit/schemas/schemas.h

```diff
@@ -176,5 +182,49 @@ class ocsf_base_event : public ocsf_base_impl {
 
         return h;
     }
+
+    /// Method to estimate the size of an ocsf message
+    ///
+    /// This works by iterating on the fields within the tuple returned by the
+    /// 'equality_fields()' method. This method is implemented on all ocsf
+    /// structures.
+    ///
+    /// The returned size is 'estimated' because it doesn't take into account
+    /// things like padding and any fields that a struct contains that it did
+    /// not include in its return value for 'equality_fields()'. Currently
+    /// the only example of this is the 'network_endpoint' struct.
+    template<typename T>
+    size_t estimated_ocsf_size(const T& t) const noexcept {
```

Comment:
why doesn't each ocsf type introduce a `size()` method?

Reply:
I would still need to iterate over all "fields" in an ocsf struct to get a total size. Also there are many ocsf types, all of them would need a size field.